tools/pd-api-bench/main.go

```diff
@@ -0,0 +1,315 @@
+// Copyright 2023 TiKV Project Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"crypto/tls"
+	"flag"
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"os/signal"
+	"strconv"
+	"strings"
+	"syscall"
+	"time"
+
+	pd "github.com/tikv/pd/client"
+	"github.com/tikv/pd/client/tlsutil"
+	"github.com/tools/pd-api-bench/cases"
+	"go.uber.org/zap"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/keepalive"
+)
+
+var (
+	pdAddr    = flag.String("pd", "127.0.0.1:2379", "pd address")
+	debugFlag = flag.Bool("debug", false, "print the output of api response for debug")
+
+	httpCases = flag.String("http-cases", "", "http api cases")
+	gRPCCases = flag.String("grpc-cases", "", "grpc cases")
+
+	client = flag.Int("client", 1, "client number")
+
+	qps   = flag.Int64("qps", 1000, "qps")
+	burst = flag.Int64("burst", 1, "burst")
+
+	// tls
+	caPath   = flag.String("cacert", "", "path of file that contains list of trusted SSL CAs")
+	certPath = flag.String("cert", "", "path of file that contains X509 certificate in PEM format")
+	keyPath  = flag.String("key", "", "path of file that contains X509 key in PEM format")
+)
+
+var base int64 = int64(time.Second) / int64(time.Microsecond)
```

Comment:
do you mean `Millisecond`?

Reply:
It is `Microsecond`, times  `Microsecond`  when calculating `tt`